### ISSUE TITLE
CUDA 8,  loss number always jumps to 87.3365 with GPU learning

### ISSUE THREAD

**Issue by Orlaiviai** [BVLC/caffe#4602]
I0818 01:17:12.052695 30915 solver.cpp:228] Iteration 20420, loss = 0.000200683
I0818 01:17:12.052738 30915 solver.cpp:244]     Train net output #0: loss = **0.000200677** (\* 1 = 0.000200677 loss)
I0818 01:17:12.052752 30915 sgd_solver.cpp:106] Iteration 20420, lr = 0.0001
I0818 01:17:15.557899 30915 solver.cpp:228] Iteration 20440, loss = **0.000163291**
I0818 01:17:15.557940 30915 solver.cpp:244]     Train net output #0: loss = 0.000163287 (\* 1 = 0.000163287 loss)
I0818 01:17:15.557948 30915 sgd_solver.cpp:106] Iteration 20440, lr = 0.0001
I0818 01:17:19.063204 30915 solver.cpp:228] Iteration 20460, loss = **6.01552e-05**
I0818 01:17:19.063244 30915 solver.cpp:244]     Train net output #0: loss = 6.01516e-05 (\* 1 = 6.01516e-05 loss)
I0818 01:17:19.063251 30915 sgd_solver.cpp:106] Iteration 20460, lr = 0.0001
I0818 01:17:22.568697 30915 solver.cpp:228] Iteration 20480, loss = **87.3365**
I0818 01:17:22.568737 30915 solver.cpp:244]     Train net output #0: loss = 87.3365 (\* 1 = 87.3365 loss)
I0818 01:17:22.568745 30915 sgd_solver.cpp:106] Iteration 20480, lr = 0.0001
I0818 01:17:26.069327 30915 solver.cpp:228] Iteration 20500, loss = **87.3365**
I0818 01:17:26.069367 30915 solver.cpp:244]     Train net output #0: loss = 87.3365 (\* 1 = 87.3365 loss)
I0818 01:17:26.069376 30915 sgd_solver.cpp:106] Iteration 20500, lr = 0.0001
I0818 01:17:29.570209 30915 solver.cpp:228] Iteration 20520, loss = 87.3365
I0818 01:17:29.570330 30915 solver.cpp:244]     Train net output #0: loss = 87.3365 (\* 1 = 87.3365 loss)
I0818 01:17:29.570339 30915 sgd_solver.cpp:106] Iteration 20520, lr = 0.0001

Loss number in GPU learning always jumps to 87.3365. Never had problem with CPU learning.
Tried many times, learning from start or from snapshot. Always problems with GPU learning after 100-700 iterations.

GeForce GTX 760
Ubuntu 16.04
Driver Version: 367.27 
GCC 5.4 / Clang 3.8
cuDNN 5.1 / cuDNN 4
CUDA 8
ATLAS / OpenBLAS
imagenet parameters

**Comment by markb729** [comment]
I have observed this phenomenon with CUDA 7 when estimates of gradients become unstable, which often happens when your batch size is small (~5) and almost always happens when batch size=1.  The option iter_size will allow you to **accumulate** an estimate of the gradient. As implemented in Caffe, iter_size will compute a cumulative sum of the gradients while iterating through each batch (i.e. iter_size * batch_size). Once complete, Caffe estimates the gradient much more accurately, weights are updated, and the iterative process continues.

**Comment by MechEngHN** [comment]
I observe the problem on my machine, too. I use Cuda 7.5 on Windows 7- 64 bit. Training a set of 3x60x60 images with batch_size = 32. 

What can possibly be the reason for this issue?

**Comment by markb729** [comment]
If not attributable to small batch sizes or sparse data causing problems for the SGD estimation, there may be an issue with image labeling. Using a LMDB backend, you must begin with label = 0 and increment by +1 for each image, i.e. 0 to num_images. I will post a new inquiry as it came up in my work and gives the "87.3365" loss issue. Only way to fix was label from 0 to num_images.

**Comment by MechEngHN** [comment]
@markb729 I don't think so. My data label was numbered in ascending order. 
I think it could be the initial weight setting or the lack of computational memory or both. Will update if I find out the reason

**Comment by Schallerwf** [comment]
Possibly related. I experienced the same issue and noticed that **my partition was full**. As the training produced snapshot files the partition would fill and loss would always be 87.3365

Made some room on my drive and the issue immediately went away.

**Comment by ujsyehao** [comment]
@Schallerwf  yeah, you are right,  when my computer memory is full, the training loss is always 87.3365
